django/db/models/aggregates.py

```diff
@@ -105,7 +105,7 @@ def resolve_expression(
     @property
     def default_alias(self):
         expressions = self.get_source_expressions()
-        if len(expressions) == 1 and hasattr(expressions[0], "name"):
+        if len(list(filter(None, expressions))) == 1 and hasattr(expressions[0], "name"):
```

Comment:
One issue I faced was that I couldn't really make the results of `get_source_expressions` non-variadic. It causes issues when what we need to get is the source expressions, excluding filtering and ordering. This might be easier if we merged the `OrderableAggMixin` directly into the `Aggregate` class and could know there are always exactly two extra, nullable expressions on the end, but for now, it requires taking a bit of an educated guess with things like `filter(None, expressions)` to get around the variadic nature.

I explored using an `ExpressionList` for the source expressions instead, but ran into two major issues. The first being that importing it caused a cyclic import loop. The second, bigger issue being that so much of `Func` depends on `self.source_expressions` being variadic that collapsing it into a non-iterable `ExpressionList` created a lot of problems.